cmake/OpenCogFindPython.cmake

```diff
@@ -0,0 +1,90 @@
+# ----------------------------------------------------------
+# Python and Cython
+#
+# NOTE: Python interpreter is needed for runing python unit tests,
+# and for running the FindCython module.
+#
+# Search for Python3 first, and use that, if found. Else use Python2.
+# To use Python2 only from the build directory run the following
+# rm CMakeCache.txt && cmake -DCMAKE_DISABLE_FIND_PACKAGE_Python3Interp=TRUE ..
+
+FIND_PACKAGE(Python3Interp)
+IF (3.4.0 VERSION_LESS "${PYTHON3_VERSION_STRING}")
+	SET (HAVE_PY_INTERP 1)
+	MESSAGE(STATUS "Python ${PYTHON3_VERSION_STRING} interpreter found.")
+ENDIF()
+
+IF (NOT HAVE_PY_INTERP)
+	FIND_PACKAGE(PythonInterp)
+	IF (2.7.0 VERSION_LESS ${PYTHON_VERSION_STRING})
+		SET (HAVE_PY_INTERP 1)
+		MESSAGE(STATUS "Python ${PYTHON_VERSION_STRING} interpreter found.")
+	ENDIF()
+ENDIF()
+
+FIND_PACKAGE(PythonLibs)
+IF (PYTHONLIBS_FOUND AND
+     ((PYTHON3INTERP_FOUND AND 3.4.0 VERSION_LESS ${PYTHONLIBS_VERSION_STRING})
+     OR
+     (2.7.0 VERSION_LESS ${PYTHONLIBS_VERSION_STRING})))
+	SET (HAVE_PY_LIBS 1)
+	MESSAGE(STATUS "Python ${PYTHONLIBS_VERSION_STRING} libraries found.")
+ELSE()
+	MESSAGE(STATUS "Python libraries NOT found.")
+ENDIF()
+
+# Cython is used to generate python bindings.
+IF(HAVE_PY_INTERP)
+	FIND_PACKAGE(Cython 0.23.0)
```

Comment:
maybe we can use any cython newer than 0.23?

Reply:
Excuse me didn't get. Actually this `FIND_PACKAGE(Cython 0.23.0)` should find any Cython which is not older than 0.23. `EXACT` is an option which forbids using other versions and there is no `EXACT`.